### PR TITLE
Adds claim api for brave wallet

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/12313

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`, `npm run gn_check`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Requested a security/privacy review as needed.
- [x] Added appropriate QA labels (`QA/Yes` or `QA/No`) to the associated issue
- [x] Added appropriate release note labels (`release-notes/include` or `release-notes/exclude`) to the associated issue
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Review by kylehickinson — APPROVED** [review]
LGTM!

File: components/brave_rewards/browser/static_values.h [hunk 1]
```diff
@@ -22,7 +22,9 @@ const std::vector<std::string> kOnlyAnonWalletCountries = {
 };
 
 #if defined(OS_ANDROID)
-  const std::map<std::string, bool> kBoolOptions = {};
+  const std::map<std::string, bool> kBoolOptions = {
+      {ledger::option::kClaimUGP, false}
+  };
```
Comment (emerick) [inline components/brave_rewards/browser/static_values.h:27]: It's usually better to put the declaration in the header file and the actual definition in a .cc file. Doing it the way it is here can lead to linker errors or subtle bugs.

File: components/brave_rewards/browser/static_values.h [hunk 2]
```diff
@@ -36,7 +38,9 @@ const std::vector<std::string> kOnlyAnonWalletCountries = {
       {ledger::option::kPublisherListRefreshInterval,
        7* base::Time::kHoursPerDay * base::Time::kSecondsPerHour}};
 #else
-  const std::map<std::string, bool> kBoolOptions = {};
+  const std::map<std::string, bool> kBoolOptions = {
+      {ledger::option::kClaimUGP, false}
+  };
```
Comment (emerick) [inline components/brave_rewards/browser/static_values.h:43]: Same issue as above.

File: vendor/bat-native-ledger/src/bat/ledger/internal/endpoint/promotion/post_claim_brave/post_claim_brave.cc [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    return type::Result::LEDGER_ERROR;
+  }
+
+  if (status_code == net::HTTP_NOT_FOUND) {
+    BLOG(0, "Not found");
+    return type::Result::NOT_FOUND;
+  }
+
+  if (status_code == net::HTTP_CONFLICT) {
+    BLOG(0, "Not found");
```
Comment (emerick) [inline vendor/bat-native-ledger/src/bat/ledger/internal/endpoint/promotion/post_claim_brave/post_claim_brave.cc:72]: Is this a copy and paste error from the `BLOG` above or we really want to say `Not found` here too?

File: vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_transfer.cc [hunk 4]
```diff
@@ -29,8 +29,19 @@ PromotionTransfer::PromotionTransfer(LedgerImpl* ledger) :
 
 PromotionTransfer::~PromotionTransfer() = default;
 
-void PromotionTransfer::Start(ledger::ResultCallback callback) {
-  auto tokens_callback = std::bind(&PromotionTransfer::GetEligibleTokens,
+void PromotionTransfer::GetAmount(
+    ledger::GetTransferableAmountCallback callback) {
+  GetEligibleTokens([callback](type::UnblindedTokenList list) {
+    double amount = 0.0;
+    for (auto& item : list) {
```
Comment (emerick) [inline vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_transfer.cc:36]: Can `item` be const?

File: components/brave_rewards/browser/static_values.h (hunk 1 above)
Comment (NejcZdovc) [reply components/brave_rewards/browser/static_values.h:27]: please file an issue and we can refactor, this one just uses existing flows

File: vendor/bat-native-ledger/src/bat/ledger/internal/endpoint/promotion/post_claim_brave/post_claim_brave.cc (hunk 3 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/endpoint/promotion/post_claim_brave/post_claim_brave.cc:72]: fixed

File: vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_transfer.cc (hunk 4 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_transfer.cc:36]: fixed

**Review by kylehickinson — APPROVED** [review]

File: vendor/bat-native-ledger/src/bat/ledger/internal/endpoint/promotion/post_claim_brave/post_claim_brave.cc [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+}
+
+void PostClaimBrave::Request(
+    const std::string& destination_payment_id,
+    PostClaimBraveCallback callback) {
+  auto url_callback = std::bind(&PostClaimBrave::OnRequest,
+      this,
+      _1,
+      callback);
+  const std::string& payload = GeneratePayload(destination_payment_id);
```
Comment (emerick) [inline vendor/bat-native-ledger/src/bat/ledger/internal/endpoint/promotion/post_claim_brave/post_claim_brave.cc:95]: Nit: Don't need the `&` here.

**Review by emerick — APPROVED** [review]
LGTM, just had one remaining tiny nit I didn't notice the first time.

File: vendor/bat-native-ledger/src/bat/ledger/internal/endpoint/promotion/post_claim_brave/post_claim_brave.cc (hunk 5 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/endpoint/promotion/post_claim_brave/post_claim_brave.cc:95]: removed

**Pull request merged**